libevmasm/Assembly.h

```diff
@@ -89,6 +89,8 @@ class Assembly
 	AssemblyItem newPushImmutable(std::string const& _identifier);
 	AssemblyItem newImmutableAssignment(std::string const& _identifier);
 	AssemblyItem newAuxDataLoadN(size_t offset);
+	AssemblyItem newSwapN(size_t _depth);
+	AssemblyItem newDupN(size_t _depth);
```

Comment:
```suggestion
	AssemblyItem newAuxDataLoadN(size_t offset) const;
	AssemblyItem newSwapN(size_t _depth) const;
	AssemblyItem newDupN(size_t _depth) const;
```

Just noticed that `newAuxDataLoadN()` should have been `const` (or even `static`) as well.

Reply:
As an aside, the whole group of `newXXX()` functions is a really weird design. Especially the older ones, which create an `AssemblyItem` that's not a part of the assembly but still modify its state (`m_immutables`, `m_currentCodeSection`, etc.). Using such item with any other assembly would be questionable but the caller may not even be aware.

Reply:
Yeah... I just copied it from auxdataload, but it also struck me as weird generally :-). We have a lot of weird boilerplate redundancies in all this - not sure we really need the AbstractAssembly either really... that was part of a very old plan to rewrite the evmasm part in a much simpler manner after the Yul optimizer had taken over a bulk of the heavy lifting...

Reply:
Wasn't `AbstractAssembly` also one of the abstractions introduced for Ewasm?